pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -535,14 +535,10 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 	if regional && utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
 		disk, err = gce.getRegionalDiskByName(diskName)
 		if err != nil {
-			glog.V(5).Infof("Could not find regional PD named %q to Attach. Will look for a zonal PD", diskName)
-			err = nil
-		} else {
-			mc = newDiskMetricContextRegional("attach", gce.region)
+			return err
 		}
-	}
-
-	if disk == nil {
```

Comment:
Why remove `if disk == nil {`?

Reply:
Before, if a regional PD is not found, we let the rest of the function continue to execute, so the `if disk == nil` is there to check whether a regional PD has been found.

Now, the function returns right away if an error is found, and `disk` is always nil if `gce.getDiskByName()` gets called